mapcompose/src/main/java/ovh/plrapps/mapcompose/api/LayoutApi.kt

```diff
@@ -190,6 +197,48 @@ suspend fun MapState.scrollTo(
     }
 }
 
+/**
+ * Scrolls to an area, animating the scroll and the scale. The target position will be centered
+ * on the area, scaled in as much as possible while still keeping the area plus the provided
+ * padding completely in view.
+ *
+ * @param area The [BoundingBox] of the target area to scroll to.
+ * @param padding Padding around the area defined as a fraction of the viewport.
+ * @param animationSpec The [AnimationSpec]. Default is [SpringSpec] with low stiffness.
+ */
+suspend fun MapState.scrollTo(
+    area: BoundingBox,
+    padding: Offset = Offset(0f, 0f),
+    animationSpec: AnimationSpec<Float> = SpringSpec(stiffness = Spring.StiffnessLow),
+) {
+    with(zoomPanRotateState) {
+        awaitLayout()
+
+        val centerX = (area.xLeft + area.xRight) / 2
+        val centerY = (area.yTop + area.yBottom) / 2
+
+        val xAxisScale = fullHeight / fullWidth.toDouble()
+        val normalizedArea = area.scaleAxis(1 / xAxisScale)
+        val rotatedNormalizedArea = normalizedArea.rotate(Point(centerX / xAxisScale, centerY), -rotation.toRad())
+        val rotatedArea = rotatedNormalizedArea.scaleAxis(xAxisScale)
```

Comment:
We need to rotate the bounding box, and to rotate points we need a coordinate system with the same scale on both axes, so point `0,1` rotated 90⁰ clockwise centered around `0,0` ends up at `1,0`. Because both axes are in the range `0..1`, but the full image is not necessarily a square, the axes are not in the same scale, which would mess up our rotations. To fix this, I am scaling the X axis so that it's in the same unit as the Y axis. It doesn't matter if I scale the X axis or the Y axis, I just need to scale one so it's in the same unit as the other. After the bounding boxes coordinate system is normalized this way, I can do my rotations, and the reverse the scaling back to the coordinate system the library uses.

In the end, both `horizontalScale` and `verticalScale` are potentially affected, but this is not very related.

Reply:
Alright! I wasn't sure, but now it's clear that it's intentional. 
I would just put `scaleAxis` as internal, because the name isn't explicit (it scales the X axis, and there's no other function to scale Y axis). Or maybe you could do without creating that function -- your call.

Reply:
That made me realize that I was wrong in the `MapView.visibleArea` API that I added yesterday. I was able to fix it by applying the same idea. Thanks!

Reply:
Looking at the full signature you can see it's `scaleAxis(xAxisMultiplier: Double)` so I don't think the name is terribly unclear, but you are right it should be `internal`, updated!

Reply:
Great! I have to go out now. I'll be back in ~3hours.